src/components/commands.js

```diff
@@ -0,0 +1,36 @@
+/**
+ * Index all commands
+ * The name of the file will be used as the command name
+ * Commands can be multiple directories deep
+ * @type {Object}
+ */
+
+const dirreq = require('../internal/dir-require')
+const Command = require('../classes/Command')
+const commands = dirreq('src/commands/**/*.js')
+const final = {
+  commands: {},
+  aliases: new Map()
+}
+
+Object.keys(commands).forEach(x => {
+  /* eslint-disable no-throw-literal */
+  try {
+    if (!(commands[x] instanceof Command)) throw `Command ${x} is not a WildBeast command, skipping`
+    if (final.commands[x]) throw `Can't index command ${x}, this command is duplicated, skipping`
+    final.commands[x] = commands[x]
+    if (commands[x].props.aliases && Array.isArray(commands[x].props.aliases)) {
+      commands[x].props.aliases.forEach(y => {
+        if (y.length < 1) throw `Aliases must be at least 1 character, an alias from ${x} is not, skipping`
+        if (final.commands[y]) throw `Can't use ${y} as an alias, there's a command with this name, skipping`
+        if (final.aliases.has(y)) throw `Can't set ${y} as an alias of ${x}, this alias already exists, skipping`
+        final.aliases.set(y, x)
+      })
```

Comment:
This might just be personal preference (Or Uncle Bob-ism), but I feel like these short variable names are really not all that helpful. For instance, look at the `final.aliases.set` call, that is just pants. Consider the following:
```suggestion
      commands[x].props.aliases.forEach(alias => {
        if (alias.length < 1) throw `Aliases must be at least 1 character, an alias from ${x} is not, skipping`
        if (final.commands[alias]) throw `Can't use ${alias} as an alias, there's a command with this name, skipping`
        if (final.aliases.has(alias)) throw `Can't set ${alias} as an alias of ${x}, this alias already exists, skipping`
        final.aliases.set(alias, x)
      })
```